File: app/src/main/java/ru/hse/control_system_v2/data/ButtonItemType.java
package ru.hse.control_system_v2.data;

import ru.hse.control_system_v2.R;
import ru.hse.control_system_v2.MainActivity;

public class ButtonItemType implements ItemType {

    String textInfo;
    final String IMAGE_TYPE = "button_type";

    public ButtonItemType(MainActivity ma) {
        this.textInfo = ma.getResources().getString(R.string.button_add_device);
    }

    @Override
    public int getDevId() {
        return -1;
    }

    @Override
    public String getImageType() {
        return IMAGE_TYPE;
    }

    @Override
    public String getTextInfo() {
        return textInfo;
    }
}

File: app/src/main/java/ru/hse/control_system_v2/data/classes/ButtonItemType.kt
package ru.hse.control_system_v2.data.classes

import ru.hse.control_system_v2.data.classes.device.model.ItemType

/**
 * Ui класс элемента кнопки для добавления устройства на главном экране
 */
class ButtonItemType(
    override var id: Int,
    override var name: String
): ItemType